The javadoc from Sun JDK 1.4.x fails for @value lines after @since
with a NullPointerException in ValueTaglet.toString.
This patch switches the lines in question.
It also ensures that javadoc errors won't go unnoticed but cause the build
to fail instead.

2009-02-18 Martin von Gagern

References: https://bugs.gentoo.org/217102

Index: apache-ant-1.7.1/src/main/org/apache/tools/ant/MagicNames.java
===================================================================
--- apache-ant-1.7.1.orig/src/main/org/apache/tools/ant/MagicNames.java
+++ apache-ant-1.7.1/src/main/org/apache/tools/ant/MagicNames.java
@@ -143,22 +143,22 @@ public final class MagicNames {
     /**
      * property that provides the default value for javac's and
      * javadoc's source attribute.
-     * @since Ant 1.7
      * Value: {@value}
+     * @since Ant 1.7
      */
     public static final String BUILD_JAVAC_SOURCE = "ant.build.javac.source";
 
     /**
      * property that provides the default value for javac's target attribute.
-     * @since Ant 1.7
      * Value: {@value}
+     * @since Ant 1.7
      */
     public static final String BUILD_JAVAC_TARGET = "ant.build.javac.target";
 
     /**
      * Name of the magic property that controls classloader reuse.
-     * @since Ant 1.4.
      * Value: {@value}
+     * @since Ant 1.4.
      */
     public static final String REFID_CLASSPATH_REUSE_LOADER = "ant.reuse.loader";
 
Index: apache-ant-1.7.1/build.xml
===================================================================
--- apache-ant-1.7.1.orig/build.xml
+++ apache-ant-1.7.1/build.xml
@@ -1483,6 +1483,7 @@
       locale="en"
       windowtitle="${Name} API"
       doctitle="${Name}"
+      failonerror="true"
       verbose="${javadoc.verbose}">
 
       <packageset dir="${java.dir}"/>
@@ -1510,6 +1511,7 @@
     <mkdir dir="${build.tests.javadocs}"/>
     <javadoc useexternalfile="yes"
       destdir="${build.tests.javadocs}"
+      failonerror="true"
       author="true"
       version="true"
       locale="en"
